File: gcc/testsuite/c-c++-common/gomp/scan-5.c
int
foo (int *a, int *b)
{
  int r = 0;
  #pragma omp parallel for reduction (inscan, +:r) default(none) firstprivate (a, b)
  for (int i = 0; i < 64; i++)
    {
      r += a[i];
      #pragma omp scan inclusive (r)	/* { dg-message "sorry, unimplemented: '#pragma omp scan' not supported yet" } */
      b[i] = r;
    }
  return r;
}

File: gcc/testsuite/gfortran.dg/gomp/scan-5.f90
! { dg-do compile }
! { dg-additional-options "-fdump-tree-original" }

integer function foo(a,b, n) result(r)
  implicit none
  integer :: a(n), b(n), n, i
  r = 0
  !$omp parallel do reduction (inscan, +:r) default(none) firstprivate (a, b)
  do i = 1, n
    r = r + a(i)
    !$omp scan inclusive (r)
    b(i) = r
  end do
end

! { dg-final { scan-tree-dump-times "#pragma omp parallel firstprivate\\(a\\) firstprivate\\(b\\) shared\\(r\\) default\\(none\\)" 1 "original" } }
! { dg-final { scan-tree-dump-times "#pragma omp for reduction\\(inscan,\\\+:r\\) nowait" 1 "original" } }
! { dg-final { scan-tree-dump-times "#pragma omp scan inclusive\\(r\\)" 1 "original" } }
